### PR TITLE
devtools/onion: fixed argument passing for generate and decode methods

### PR DESCRIPTION
**Pull request by remyers**
Fixed 'generate' and 'decode' methods to properly parse argv arguments.
Fixed so methods correctly use the associated data set by the --assoc-data flag.

I also added some examples to -h to better show how these methods work with real onion data.

Introduced a behavioral change to read the onion from a file instead of stdin; this makes it easier to debug in some IDEs (eg. vscode).

### CODE REVIEW

File: devtools/onion.c [hunk 1]
```diff
@@ -109,7 +109,7 @@ static void do_generate(int argc, char **argv,
 	u8 *serialized = serialize_onionpacket(ctx, res);
 	if (!serialized)
 		errx(1, "Error serializing message.");
-	printf("%s\n", tal_hex(ctx, serialized));
+	printf("%s", tal_hex(ctx, serialized));
```
Comment (rustyrussell) [inline devtools/onion.c]: Hmm, this is a bit awkward if you run it on the cmdline though?

Can we instead fix it up on the input side?

Comment (remyers) [reply devtools/onion.c]: Good point; I've rebased to leave the trailing newline but instead trim whitespace with 6a078de . Is this better?

**Comment by rustyrussell** [comment]
Nice work!

**Comment by remyers** [comment]
Thanks! and ht to @willcl-ark for the testing that lead to these fixes.

**Pull request merged**